src/battle-log.ts

```diff
@@ -90,20 +90,23 @@ class BattleLog {
 			}
 			break;
 
-		case 'join': case 'j':
-			divHTML = '<small>' + BattleLog.escapeHTML(args[1]) + ' joined.</small>';
+		case 'join': case 'j': {
+			const user = BattleTextParser.parseNameParts(args[1]);
+			divHTML = '<small>' + BattleLog.escapeHTML(user.name) + ' joined.</small>';
 			break;
-
-		case 'leave': case 'l':
-			divHTML = '<small>' + BattleLog.escapeHTML(args[1]) + ' left.</small>';
+		}
+		case 'leave': case 'l': {
+			const user = BattleTextParser.parseNameParts(args[1]);
+			divHTML = '<small>' + BattleLog.escapeHTML(user.name) + ' left.</small>';
 			break;
-
-		case 'name': case 'n':
-			if (toId(args[2]) !== toId(args[1])) {
-				divHTML = '<small>' + BattleLog.escapeHTML(args[1]) + ' renamed from ' + BattleLog.escapeHTML(args[2]) + '.</small>';
+		}
+		case 'name': case 'n': {
+			const user = BattleTextParser.parseNameParts(args[1]);
+			if (toId(args[2]) !== toId(user.name)) {
+				divHTML = '<small>' + BattleLog.escapeHTML(user.name) + ' renamed from ' + BattleLog.escapeHTML(args[2]) + '.</small>';
```

Comment:
I should note - this is going to look kind of weird once the server code lands, as `args[2]` is the ID, which shouldn't have actually changed, so its going to be `'Foo renamed from foo'` when `Foo@!Away -> Foo`.

Reply:
This should probably be a different message in that case; something like `Foo changed status to ...`